lib/Adapter/Solarium/SolariumAdapter.php

```diff
@@ -44,12 +43,6 @@ public function getNbResults(): int
         return $this->getResultSet()->getNumFound();
     }
 
-    /**
-     * @param int<0, max> $offset
-     * @param int<0, max> $length
-     *
-     * @return iterable<array-key, T>
```

Comment:
instead of updating those phpdoc types (replacing `T` with `DocumentInterface`), I chose to remove it entirely. They are not necessary as phpstan will properly inherit this info from the AdapterInterface.